server/src/main/java/io/unitycatalog/server/service/VolumeService.java

```diff
@@ -2,53 +2,176 @@
 
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
-import com.linecorp.armeria.server.annotation.*;
+import com.linecorp.armeria.server.annotation.Delete;
+import com.linecorp.armeria.server.annotation.ExceptionHandler;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.Param;
+import com.linecorp.armeria.server.annotation.Patch;
+import com.linecorp.armeria.server.annotation.Post;
+import io.unitycatalog.server.auth.UnityCatalogAuthorizer;
+import io.unitycatalog.server.auth.annotation.AuthorizeExpression;
+import io.unitycatalog.server.auth.annotation.AuthorizeKey;
+import io.unitycatalog.server.auth.annotation.AuthorizeKeys;
+import io.unitycatalog.server.auth.decorator.UnityAccessEvaluator;
 import io.unitycatalog.server.exception.GlobalExceptionHandler;
+import io.unitycatalog.server.model.CatalogInfo;
 import io.unitycatalog.server.model.CreateVolumeRequestContent;
+import io.unitycatalog.server.model.ListVolumesResponseContent;
+import io.unitycatalog.server.model.SchemaInfo;
 import io.unitycatalog.server.model.UpdateVolumeRequestContent;
+import io.unitycatalog.server.model.VolumeInfo;
+import io.unitycatalog.server.persist.CatalogRepository;
+import io.unitycatalog.server.persist.MetastoreRepository;
+import io.unitycatalog.server.persist.SchemaRepository;
 import io.unitycatalog.server.persist.VolumeRepository;
+import io.unitycatalog.server.persist.model.Privileges;
+import io.unitycatalog.server.utils.IdentityUtils;
+import lombok.SneakyThrows;
+
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.UUID;
+
+import static io.unitycatalog.server.model.SecurableType.CATALOG;
+import static io.unitycatalog.server.model.SecurableType.METASTORE;
+import static io.unitycatalog.server.model.SecurableType.SCHEMA;
+import static io.unitycatalog.server.model.SecurableType.VOLUME;
 
 @ExceptionHandler(GlobalExceptionHandler.class)
 public class VolumeService {
   private static final VolumeRepository VOLUME_REPOSITORY = VolumeRepository.getInstance();
+  private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final CatalogRepository CATALOG_REPOSITORY = CatalogRepository.getInstance();
 
-  public VolumeService() {}
+  private final UnityCatalogAuthorizer authorizer;
+  private final UnityAccessEvaluator evaluator;
+
+  @SneakyThrows
+  public VolumeService(UnityCatalogAuthorizer authorizer) {
+    this.authorizer = authorizer;
+    evaluator = new UnityAccessEvaluator(authorizer);
+  }
 
   @Post("")
-  public HttpResponse createVolume(CreateVolumeRequestContent createVolumeRequest) {
+  // TODO: for now, we are not supporting CREATE VOLUME or CREATE EXTERNAL VOLUME privileges
+  @AuthorizeExpression("""
+          #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #schema, OWNER, USE_SCHEMA)
```

Comment:
As for tables and functions, we should add the missing `CREATE FUNCTION`, `CREATE TABLE`, and `CREATE VOLUME`, and make this rule the following:

```
(#authorize(#principal, #catalog, USE_CATALOG) && #authorize(#principal, #schema, OWNER)) ||
(#authorize(#principal, #catalog, USE_CATALOG) && #authorizeAll(#principal, #schema, USE_SCHEMA, CREATE_VOLUME))
```